contractcourt/channel_arbitrator.go

```diff
@@ -2992,6 +2992,10 @@ func (c *ChannelArbitrator) handleBlockbeat(beat chainio.Blockbeat) error {
 	// Notify we've processed the block.
 	defer c.NotifyBlockProcessed(beat, nil)
 
+	// Perform a non-blocking read on the close events in case the channel
+	// is closed in this blockbeat.
+	c.receiveAndProcessCloseEvent()
```

Comment:
Should we maybe check if the state is already in contractClosed so there is no need to call this again ?

Reply:
yeah i like it, lemme try

Reply:
building in a side branch, will push once it passes all the tests